aws/lambda/pytorch-auto-revert/pytorch_auto_revert/signal.py

```diff
@@ -0,0 +1,357 @@
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+from typing import List, Optional, Set, Tuple, Union
+
+
+class SignalStatus(Enum):
+    """signal status enum"""
+
+    PENDING = "pending"
+    SUCCESS = "success"
+    FAILURE = "failure"
+
+
+# data classes:
+
+
+@dataclass
+class AutorevertPattern:
+    """
+    Represents an autorevert pattern detected in a signal.
+    """
+
+    pattern_detected: bool
+    workflow_name: str
+    newer_commits: List[str]
+    older_commit: str
```

Comment:
NIT

Not a big fan on this newer_commits, older_commit convention. Can we make like (hud seq, newer top):
newer_commit - a array only if this can be more than one, I suspect not.
suspected_commit
older_commit

Reply:
thanks, consolidated